src/examples/component-definition/json/ocpv4-example-component-definition.json

```diff
@@ -0,0 +1,209 @@
+          "type": "organization",
+          "name": "RedHat",
+          "links": [
+            {
+              "href": "https://docs.openshift.com/",
+              "rel": "website"
+            }
+          ]
+        }
+      ]
+    },
+    "rules": [
+      {
+        "name": "OpenShift Container Platform v4 Kubelet Eviction Threshhold Settings",
+        "uuid": "a969feae-ae93-422f-8a95-9362adef826b",
+        "title": "Check hardening guidelines for OCP kubelet instances with OpenSCAP",
+        "condition": "kubelet is hardened with CIS OCPv4 baseline validated with OpenSCAP",
+        "condition-tests": [
+          {
+            "uuid": "02970ddf-68f2-4bf8-8f2d-cc511766a3e8",
+            "title": "OpenSCAP | CIS | OCPv4 Baseline Tests and Parameters",
+            "description": "The `oscap` program is a command line tool that allows users to load, scan, validate, edit, and export SCAP documents.\n\nThis baseline is to test against the recommendations from the Center for Internet Security for hardening one or more instances of the OpenShift Container Platform (v4).",
+            "props": [
+              {
+                "name": "runtime",
+                "ns": "https://www.open-scap.org",
+                "value": "#9c1d4058-127e-4314-b25d-f042182eea7d",
+                "class": "dependency"
+              },
+              {
+                "name": "baselines-archive",
+                "ns": "https://www.open-scap.org",
+                "value": "#96c0023a-3ba7-4160-8dfc-e6c230885586",
+                "class": "dependency"
+              },
+              {
+                "name": "baseline-file",
+                "ns": "https://www.open-scap.org",
+                "value": "#9c584372-0f9f-44bc-b68a-8956f7b8d4f6",
+                "class": "dependency"
+              }
+            ],
+            "links": [
+              {
+                "href": "#96c0023a-3ba7-4160-8dfc-e6c230885586",
+                "rel": "dependency"
+              },
+              {
+                "href": "#9c1d4058-127e-4314-b25d-f042182eea7d",
+                "rel": "dependency"
+              },
+              {
+                "href": "#9c584372-0f9f-44bc-b68a-8956f7b8d4f6",
+                "rel": "dependency"
+              }
+            ],
+            "steps": [
+              {
+                "uuid": "9334fe43-4ee2-46ba-87e8-4b73f54819ab",
+                "title": "Ensure Eviction threshold Settings Are Set - evictionSoft: nodefs.available",
+                "description": "Execute the openscap test to confirm this sub-rule:\n\n`nodefs.available` setting of the `evictionSoft` section.",
+                "props": [
+                  {
+                    "name": "kubelet_eviction_thresholds_set_soft_nodefs_inodesfree",
+                    "ns": "https://www.open-scap.org",
+                    "value": "5%",
+                    "class": "argument"
+                  }
+                ],
```

Comment:
It seems specific rules in OCP4 CIS -
  # 1.3.1 Ensure that garbage collection is configured as appropriate
    - kubelet_eviction_thresholds_set_soft_memory_available
    - kubelet_eviction_thresholds_set_soft_nodefs_available
    - kubelet_eviction_thresholds_set_soft_nodefs_inodesfree
    - kubelet_eviction_thresholds_set_soft_imagefs_available

are captured as steps here. This is not the intended use. Each rule under #1.3.1 above should be represented as a separate rule in OSCAL and not as step. Each of these rule can be associated with control/statement under "implemented-requirements" and evaluated separately.

Reply:
> are captured as steps here. This is not the intended use. Each rule under 1.3.1 above should be represented as a separate rule in OSCAL and not as step. 

I had alluded to this when we met, but I think you could reasonably expect to do both. I guess I would need to find a way to explain this and show it in examples. Given a condition you want to test or verify from a CIS OpenSCAP baseline (or as they call it, a profile, different from an OSCAL profile, I know) do you want to all 100 tests in one rule or do? Do you want to have 100 rules separately?

In short you might have reasons for doing both.

>Each of these rule can be associated with control/statement under "implemented-requirements" and evaluated separately.
 
I thought we support this now, maybe the examples aren't clear. That's on me!

This can be done bidirectionally, but the feedback given in a prior meeting was the linkage would go from a `statement` in an `implemented-requirement`. In the example look for the `prop` with `@name="evidence"` in the examples and you will see a UUID reference pointing to the rule, or individual steps.

> Also, there should be a way to specify parameters for a rule. Same or similar construct used in catalogs for specifying parameters can be used here (inside a rule).

This will have to come after we handle parameters in a deeper way, in a later version of OSCAL. We have not forgotten!